packages/flutter/lib/src/material/stepper.dart

```diff
@@ -0,0 +1,665 @@
+/// text.
+///
+/// See also:
+///
+///  * [Step]
+enum StepState {
+  /// A step that displays its index in its circle.
+  indexed,
+  /// A step that displays a pencil icon in its circle.
+  editing,
+  /// A step that displays a tick icon in its circle.
+  complete,
+  /// A step that is disabled and does not to react to taps.
+  disabled,
+  /// A step that is currently having an error. e.g. the use has submitted wrong
+  /// input.
+  error
+}
+
+/// The type of the [Stepper]. This determines how the stepper is presented.
+enum StepperType {
+  /// A vertical layout of the steps with their content in-between the titles.
+  vertical,
+  /// A horizontal layout of the steps with their content below the titles.
+  horizontal
+}
+
+const TextStyle _kStepStyle = const TextStyle(
+  fontSize: 12.0,
+  color: Colors.white
+);
+
+/// A material step used in [Stepper]. The step can have a title and subtitle,
+/// an icon within its circle, some content and a state that governs its
+/// styling.
+///
+/// See also:
+///
+///  * [Stepper]
+///  * <https://material.google.com/components/steppers.html>
+class Step {
+  /// Creates a step for a [Stepper].
+  ///
+  /// The [title], [content], and [state] arguments must not be null.
+  Step({
+    @required this.title,
+    this.subtitle,
+    @required this.content,
+    this.state: StepState.indexed,
+    this.isActive: false
+  }) {
+    assert(this.title != null);
+    assert(this.content != null);
+    assert(this.state != null);
+  }
+
+  /// The title of the step. It changes color based on the [state].
+  final Widget title;
+
+  /// The subtitle of the step. It changes color based on the state.
+  ///
+  /// If null, the subtitle is not shown.
+  final Widget subtitle;
+
+  /// The content of the step that appears below the [title] and [subtitle].
+  ///
+  /// Below the content, every step has a 'continue' and 'cancel' button.
+  final Widget content;
+
+  /// The state of the step which determines the styling of its componenents
+  /// and whether steps are interactive.
+  final StepState state;
+
+  /// Whether or not the step is active. The flag only influences styling.
+  final bool isActive;
+}
+
+/// A material stepper widget that displays progress through a sequence of
+/// steps. They are particularly useful in the case of forms where one step
+/// requires the completion of another one, or where multiple steps need to be
+/// completed in order to submit the whole form.
+///
+/// The widget is a flexible wrapper. A parent class should pass [currentStep]
+/// to this widget based on some logic triggered by the three callbacks that it
+/// provides.
+///
+/// See also:
+///
+///  * [Step]
+///  * <https://material.google.com/components/steppers.html>
+class Stepper extends StatefulWidget {
+  /// Creates a stepper from a list of steps.
+  ///
+  /// This widget is not meant to be rebuilt with a different set of steps. If
```

Comment:
It's ugly but IMHO very hard to reach. You would have to recreate the stepper in the exact same position which would almost definitely be confusing for the end user.